File: languages/python/linkedlist.py
class Node:

    def __init__(self, data):
        self.data = data
        self.next = None

    def get_data(self):
        return self.data

    def get_next(self):
        return self.next

    def set_next(self, next):
        self.next = next

    def __str__(self):
        return str(self.data)

class LinkedList:

    def __init__(self):
        self.head = None

    def add(self, data):
        if not self.head:
            node = Node(data)
            self.head = node
            return
        else:
            node = self.head
            while node.get_next() is not None:
                node = node.get_next()
            new_node = Node(data)
            node.set_next(new_node)
            return

    def find(self, data):
        current_node = self.head
        while current_node:
            if current_node.get_data() == data:
                return data
            else:
                current_node = current_node.get_next()
        if current_node is None:
            print("No data in list")
        return current_node

    def remove(self, data):
        current_node = self.head
        while current_node.get_next() is not None:
            if current_node.get_data() == data:
                self.head = current_node.get_next()
                return
            else:
                if current_node.get_next().data == data:
                    current_node.set_next(current_node.get_next().get_next())
                    return
                else:
                    current_node = current_node.get_next()

    def print_list(self):
        node = self.head
        while node is not None:
            print(node.get_data())
            node = node.get_next()


li = LinkedList()
li.add(4)
li.add(3)
li.add(8)
li.add(20)
li.remove(4)
li.print_list()

File: languages/javascript/datastructures/linked-list.js
function Node(data) {
  this.data = data;
  this.next = null;
}

function LinkedList() {
  this.head = null;
}

LinkedList.prototype.append = function(data) {
  if (this.head == null) {
    this.head = new Node(data);
    return;
  }
  var current = this.head;
  while (current.next != null) {
    current = current.next;
  }
  current.next = new Node(data);
};

LinkedList.prototype.prepend = function(data) {
  if (this.head != null) {
    var current = new Node(data);
    current.next = this.head;
    this.head = current;
    return;
  }
};

LinkedList.prototype.remove = function(data) {
  if (this.head == null) return;
  if (this.head.data === data) this.head = this.head.next;

  var current = this.head;
  while (current.next != null) {
    while (current.next.data === data) {
      current.next = current.next.next;
      if (current.next == null) return;
    }
    current = current.next;
  }
};

LinkedList.prototype.printData = function() {
  var current = this.head;
  while (current) {
    console.log(current.data);
    current = current.next;
  }
};

function play() {
  linkedList = new LinkedList();

  for (var i = 0; i < 5; i++)
    linkedList.append(Math.floor(Math.random() * 10) + 1);

  linkedList.printData();
  console.log("-------------------------");
  linkedList.remove(4);
  linkedList.printData();
}
play();

//console.log(JSON.stringify(linkedList, null, 4));
